pkg/virt-handler/device-manager/usb_device.go

```diff
@@ -565,8 +530,14 @@ func NewUSBDevicePlugin(resourceName string, pluginDevices []*PluginDevices) *US
 			resourceName: resourceName,
 			initialized:  false,
 			lock:         &sync.Mutex{},
+			health:       make(chan deviceHealth),
+			done:         make(chan struct{}),
+			deregistered: make(chan struct{}),
 		},
 		devices: pluginDevices,
 		logger:  log.Log.With("subcomponent", resourceID),
 	}
+	usb.registerFunc = usb.registerServer
```

Comment:
I would move it inside the USBDevicePlugin declaration and it can simply written as:
```golang
	usb := &USBDevicePlugin{
		DevicePluginBase: &DevicePluginBase{
			socketPath:   SocketPath(resourceID),
			resourceName: resourceName,
			initialized:  false,
			lock:         &sync.Mutex{},
			health:       make(chan deviceHealth),
			done:         make(chan struct{}),
			deregistered: make(chan struct{}),
                        registerFunc: usb.registerServer
                        healthcheck: usb.healthCheck
		},
		devices: pluginDevices,
		logger:  log.Log.With("subcomponent", resourceID),
	}
```

Reply:
Nevermind, my bad eyes. I just noticed that you need to initialize device internals

Reply:
indeed, the function can be referenced as dpi.registerFunc, registerFunc alone is undefined

Reply:
However, it is still looks a bit wrong to me. I'm thinking to a nicer way to rewrite this

Reply:
Yeah its a bit unclean i agree but I think its the only way you can do this
You can for example make the function not defined on the plugin and pass the plugin as an argument.
In this case you will need to pass a reference to the parent plugin into the base plugin so it executes the function with the main plugin as a parameter, hence creating a struct embedding infinite loop

If you have any ideas though let me know